server/shadow/shadow_client.c

```diff
@@ -30,10 +30,62 @@
 
 #include <freerdp/log.h>
 
+#include "../../libfreerdp/core/server.h"
 #include "shadow.h"
 
 #define TAG CLIENT_TAG("shadow")
 
+struct _SHADOW_GFX_STATUS
+{
+	BOOL gfxOpened;
+	BOOL gfxSurfaceCreated;
+};
+typedef struct _SHADOW_GFX_STATUS SHADOW_GFX_STATUS;
+
+static void shadow_client_rdpgfx_new_surface(rdpShadowClient *client)
+{
```

Comment:
I guess this function should not be void as it calls some callbacks that can return errors